src/runtime/float.cpp

```diff
@@ -926,6 +926,22 @@ template <ExceptionStyle S> Box* floatNew(BoxedClass* _cls, Box* a) noexcept(S =
     return new (cls) BoxedFloat(f->d);
 }
 
+// Roughly analogous to CPython's float_new.
+// The arguments need to be unpacked from args and kwds.
+static Box* floatNewPacked(BoxedClass* type, Box* args, Box* kwds) noexcept {
+    PyObject* x = False;
+    static char* kwlist[2] = { NULL, NULL };
+    kwlist[0] = const_cast<char*>("x");
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "|O:float", kwlist, &x))
+        return NULL;
+
+    if (x == NULL)
+        return floatNew<CAPI>(type, None);
```

Comment:
I think the arg should be ```boxFloat(0.0)``` not ```None```.
But than I think the ```NULL``` check will never trigger because ```x``` is initialized to ```False``` and marked as optional inside ```PyArg_ParseTupleAndKeywords``` .
I think we could just get rid of the if check in cpython does and just call all the time into ```floatNew<CAPI>(type, x); ```  and similar to cpython make use of the fact that if ```x``` is ```False``` and ```floatNew``` will create a ```0.0``` float for it.